### PR TITLE
Add new tests for alerts (DeadManSnitch and SLO)

### PR DESCRIPTION
**Pull request by jgarciao**
- Adds 5 tests related to alerts
- 4 of them included in Tier3 because each need more than 20 minutes to run
- ODS-738 is marked as Skip because it's an alert that was disabled (I think temporally) in RHODS 1.3.0

### CODE REVIEW

**Comment by jgarciao** [comment]
It's being tested right not here: https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/443

**Comment by jgarciao** [comment]
Tested successfully here: https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/446/

File: tests/Resources/Page/ODH/Prometheus/Prometheus.robot [hunk 1]
```diff
@@ -68,34 +68,59 @@ Alert Should Be Firing     # robocop: disable:too-many-calls-in-keyword
...
+        Fail    msg=Alert "${alert} ${alert-duration}" was not found in Prometheus firing rules
+    END
+
+Alerts Should Be Equal
+    [Documentation]    Compares two alerts names and fails if they are different.
+    ...    If ${alert1-duration} is not empty, compare it also with ${alert2-duration}
+    [Tags]    Private
+    [Arguments]    ${alert1-name}    ${alert1-duration}    ${alert2-name}    ${alert2-duration}
+    Should Be Equal    ${alert1-name}    ${alert2-name}
+    IF    "${alert1-duration}" != "${EMPTY}"
```
Comment (bdattoma) [inline tests/Resources/Page/ODH/Prometheus/Prometheus.robot:100]: what if alert1 duration is Empty? Doesn't matter if alert 2 duration is not empty in that case?

File: tests/Tests/200__monitor_and_manage/200__metrics/203__alerts.robot [hunk 2]
```diff
@@ -10,18 +10,38 @@ Suite Setup         Alerts Suite Setup
...
 ...                                     203__alerts/notebook-pvc-usage/fill-notebook-pvc-over-90.ipynb
 
-${TEST_ALERT_PVC100_NOTEBOOK_PATH}=     SEPARATOR=
+${TEST_ALERT_PVC100_NOTEBOOK_PATH}      SEPARATOR=
 ...                                     /ods-ci-notebooks-main/notebooks/200__monitor_and_manage/
 ...                                     203__alerts/notebook-pvc-usage/fill-notebook-pvc-to-100.ipynb
 
 
 *** Test Cases ***
+Verify RHODS Prometheus Alerts Are Not Firing Except For DeadManSnitch
```
Comment (bdattoma) [inline tests/Tests/200__monitor_and_manage/200__metrics/203__alerts.robot:25]: just a suggestion for the name  `Verify RHODS Prometheus Alerts Fire For DeadManSnitch Only`

File: tests/Tests/200__monitor_and_manage/200__metrics/203__alerts.robot [hunk 3]
```diff
@@ -47,9 +68,134 @@ Verify Alert RHODS-PVC-Usage-At-100 Is Fired When User PVC Is At 100 Percent
...
+    ...    SLOs-haproxy_backend_http_responses_total
+    ...    RHODS Route Error Burn Rate
+    ...    alert-duration=120
+
+    ODS.Scale Deployment    redhat-ods-operator    rhods-operator    replicas=0
+    ODS.Scale Deployment    redhat-ods-applications    traefik-proxy    replicas=0
+
+    Prometheus.Wait Until Alert Is Firing    ${RHODS_PROMETHEUS_URL}
+    ...    ${RHODS_PROMETHEUS_TOKEN}
+    ...    SLOs-haproxy_backend_http_responses_total
```
Comment (bdattoma) [inline tests/Tests/200__monitor_and_manage/200__metrics/203__alerts.robot:92]: I'd explicitly write the argument names for readability purposes

**Comment by bdattoma** [comment]
LGTM. I've left a couple of small comments

**Review by bdattoma — APPROVED** [review]

File: tests/Resources/Page/ODH/Prometheus/Prometheus.robot (hunk 1 above)
Comment (jgarciao) [reply tests/Resources/Page/ODH/Prometheus/Prometheus.robot:100]: Yes, it doesn't mater in that case. This is used for DeadManSnitch and other alerts, that don't have duration defined (or is = 0)

File: tests/Tests/200__monitor_and_manage/200__metrics/203__alerts.robot (hunk 3 above)
Comment (jgarciao) [reply tests/Tests/200__monitor_and_manage/200__metrics/203__alerts.robot:92]: I've added it in some parameters where there could be confusion (like alert-duration and timeout) but I prefer not to add it to all of them, at least not in this PR

File: tests/Tests/200__monitor_and_manage/200__metrics/203__alerts.robot [hunk 4]
```diff
@@ -10,18 +10,38 @@ Suite Setup         Alerts Suite Setup
...
+${TEST_ALERT_PVC100_NOTEBOOK_PATH}      SEPARATOR=
 ...                                     /ods-ci-notebooks-main/notebooks/200__monitor_and_manage/
 ...                                     203__alerts/notebook-pvc-usage/fill-notebook-pvc-to-100.ipynb
 
 
 *** Test Cases ***
+Verify RHODS Prometheus Alerts Are Not Firing Except For DeadManSnitch
+    [Documentation]    Verifies that, in a regular situation, only the DeadManSnitch alert is firing
+    [Tags]    Smoke    Tier1    ODS-540
+
```
[the remark this answers is not in the document]
Comment (jgarciao) [reply tests/Tests/200__monitor_and_manage/200__metrics/203__alerts.robot:28]: Done

**Comment by jgarciao** [comment]
Testing again after changes here: https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/459

File: tests/Tests/200__monitor_and_manage/210__alerts/210__alerts.robot [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+    [Tags]    Tier3
+    ...       ODS-738
+
+    Skip    msg=This alert was disabled in RHODS 1.3.0. More info at RHODS-2101
+
+    Skip Test If Alert Is Already Firing    ${RHODS_PROMETHEUS_URL}
+    ...    ${RHODS_PROMETHEUS_TOKEN}
+    ...    SLOs-haproxy_backend_http_responses_total
+    ...    RHODS Route Error Burn Rate
+    ...    alert-duration=120
```
Comment (tarukumar) [inline tests/Tests/200__monitor_and_manage/210__alerts/210__alerts.robot:90]: alert duration we should add unit of time to the number like _120s****_

Comment (jgarciao) [reply tests/Tests/200__monitor_and_manage/210__alerts/210__alerts.robot:90]: This value is not a Robot Framework timeout, it is matching a value returned by the prometheus API. I prefer to leave it like this

**Review by tarukumar — APPROVED** [review]
LGTM

**Review by bdattoma — APPROVED** [review]
lgtm

**Pull request merged**